components/payments/cmd/connectors/internal/connectors/modulr/task_fetch_transactions.go

```diff
@@ -19,12 +19,29 @@ import (
 	"go.opentelemetry.io/otel/attribute"
 )
 
+type fetchTransactionsState struct {
+	LastTransactionDate string `json:"last_transaction_date"`
+}
+
+func (s *fetchTransactionsState) SetLatest(latest *client.Transaction) error {
+	lastTransactionTime, err := time.Parse("2006-01-02T15:04:05.999-0700", latest.TransactionDate)
+	if err != nil {
+		return err
+	}
+	tnDate := lastTransactionTime.Format("2006-01-02T15:04:05-0700")
+	if s.LastTransactionDate == "" || tnDate > s.LastTransactionDate {
```

Comment:
Same comment here for the string comparaison

Reply:
Fixed too